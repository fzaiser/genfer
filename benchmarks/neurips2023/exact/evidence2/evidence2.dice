let evidence = flip 0.5 in
    if evidence then
        let coin1  = flip 0.5 in
        let tmp = observe coin1 in
        coin1
    else
        flip 0.5