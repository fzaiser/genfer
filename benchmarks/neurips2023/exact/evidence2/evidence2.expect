Support is a subset of: {0, ..., 1}

Computing moments...
Total measure:             Z = 0.75
Expected value:            E = 0.6666666666666666
2nd raw moment:         μ'_2 = 0.6666666666666666
3rd raw moment:         μ'_3 = 0.6666666666666666
4th raw moment:         μ'_4 = 0.6666666666666666
Standard deviation:        σ = 0.4714045207910317
Variance (2nd central):    V = 0.2222222222222222
3rd central moment:      μ_3 = -0.07407407407407413
4th central moment:      μ_4 = 0.07407407407407368
Skewness (3rd std moment): S = -0.707106781186548
Kurtosis (4th std moment): K = 1.4999999999999922

Computing probabilities up to 2...
Unnormalized: p(0)     = 0.25
Normalized:   p(0) / Z = 0.3333333333333333
Unnormalized: p(1)     = 0.5
Normalized:   p(1) / Z = 0.6666666666666666
Unnormalized: p(n)     <= 0.0 for all n >= 2
Normalized:   p(n) / Z <= 0.0 for all n >= 2
