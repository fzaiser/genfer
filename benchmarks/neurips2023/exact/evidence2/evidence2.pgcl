nat evidence;
nat coin;

evidence := bernoulli(0.5);
coin := 0;
if(evidence = 1) {
    coin1 := bernoulli(0.5);
    observe(coin1 = 1);
    coin := coin1;
} else {
    coin := bernoulli(0.5);
}

?Pr[coin];
