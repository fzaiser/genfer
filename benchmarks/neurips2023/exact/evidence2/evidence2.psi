// flags: --dp
def main(){
    evidence := flip(1/2);
    coin := 0;
    if evidence {
        coin1 := flip(1/2);
        observe(coin1);
        coin = coin1;
    }else{
        coin = flip(1/2);
    }
    return coin; // expected: 1/3·δ(0)[coin]+2/3·δ(1)[coin]
}
