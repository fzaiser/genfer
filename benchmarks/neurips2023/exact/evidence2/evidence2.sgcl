evidence ~ Bernoulli(0.5);
coin := 0;
if evidence = 1 {
    coin1 ~ Bernoulli(0.5);
    observe coin1 = 1;
    coin := coin1;
} else {
    coin ~ Bernoulli(0.5);
}
return coin;