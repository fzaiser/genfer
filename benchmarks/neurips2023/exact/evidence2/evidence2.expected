0.66666666
p(1) / Z = 2/3
2/3·δ(1)[coin]
2/3*coin
